pkg/pvc/pvc_protected_entity.go

```diff
@@ -283,6 +290,72 @@ func (this PVCProtectedEntity) getProtectedEntityForPV(ctx context.Context, pv *
 				return nil, errors.Errorf("Unexpected access mode, %v, for Persistent Volume %s", pv.Spec.AccessModes[0], pv.Name)
 			}
 		}
+	} else if pv.Spec.VsphereVolume != nil {
+		if pv.Spec.AccessModes[0] != core_v1.ReadWriteOnce {
+			return nil, errors.Errorf("Unexpected access mode, %v, for Persistent Volume %s", pv.Spec.AccessModes[0], pv.Name)
+		}
+		if pv.Annotations["pv.kubernetes.io/migrated-to"] == VSphereCSIProvisioner {
+			volumePath := pv.Spec.VsphereVolume.VolumePath
+
+			// Migrated volume must be of type ivd
+			pvPEType := this.getComponentPEType()
+
+			var config *rest.Config
+			path := os.Getenv("KUBECONFIG")
+			config, err := clientcmd.BuildConfigFromFlags("", path)
+			client, err := dynamic.NewForConfig(config)
+			if err != nil {
+				return nil, errors.Wrapf(err, "Could not set up client for config")
+			}
+			migrationGVR := schema.GroupVersionResource{Group: "cns.vmware.com", Version: "v1alpha1", Resource: "cnsvspherevolumemigrations"}
+			migrationResource := client.Resource(migrationGVR)
+			migrationResourcelist, err := migrationResource.List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return nil, errors.Wrapf(err, "Could not list resource for cnsvspherevolumemigrations")
+			}
+			var pvIDstr string
+			for _, item := range migrationResourcelist.Items {
+				name := item.GetName()
+				// Parse volumeid and volumepath field from cnsvspherevolumemigrations
+				migrationVolumeId, exists, err := unstructured.NestedString(item.UnstructuredContent(), "spec", "volumeid")
+				if err != nil {
+					this.logger.Infof("Error reading cnsvspherevolumemigrations %s: %v", name, err)
+					continue
+				}
+				if !exists {
+					migrationVolumeId = "NOT-FOUND"
```

Comment:
I still have concern about "NOT-FOUND" here. If either migrationVolumeId or migrationVolumePath doesn't exist, we can just continue the loop, right? why do we have to set some value, like, "NOT-FOUND" to variables.

At the end of the for lop, if `pvIDstr` is still empty string, it is better to error out and notify users about it.